### PR TITLE
Focused Launch: Split Summary into modular Steps

### PR DESCRIPTION
**Pull request by ciampo**
## Changes proposed in this Pull Request

* Refactor `<Summary />` component and split it into different steps.
* Introduce the concept of `activeSteps`(steps that require user interaction)  vs `disabledSteps` (steps that are not interactive, and only serve as a summary to the user)

These changes are part of the preparation work that facilitates the changes required by #46858

## Testing instructions

Since this PR is a refactor, it's important to test that the Focused Launch modal as a whole still works like in production.

### Compare the in-editor versions (local vs production)

* Make sure you have an active connection to your sandbox, and that your sandbox is clean and up to date
* In two separate terminal windows, from the root of the project, run:
  - `yarn start`
  - `cd apps/editing-toolkit && yarn dev --sync`
* Sandbox an unlaunched WordPress site created via `/start` by adding it to your host file
* Visit the URL `/page/[SITE_ID]/home?flags=create/focused-launch-flow` on your local machine, and compare it to the production version, by visiting the same path on `wordpress.com`. There shouldn't be any difference between the local and the production version

### Check that the domain step renders on top of the other steps, when the site has already got an associated paid domain

* Make sure you have an active connection to your sandbox, and that your sandbox is clean and up to date
* In two separate terminal windows, from the root of the project, run:
  - `yarn start`
  - `cd apps/editing-toolkit && yarn dev --sync`
* Sandbox an unlaunched WordPress site created via `/start` by adding it to your host file. 
* **Add a paid domain to the site**
* Visit the URL `/page/[SITE_ID]/home?flags=create/focused-launch-flow` on your local machine, and compare it to the production version, by visiting the same path on `wordpress.com`. There shouldn't be any difference between the local and the production version

See https://github.com/Automattic/wp-calypso/pull/47125/files#r518118936 for more details

## Screenshots

Focused Launch, when the site doesn't have a paid domain already associated:

Screenshot 2020-11-05 at 16 11 36

Focused Launch, when the site already has a paid domain associated:

Screenshot 2020-11-05 at 16 08 52

### CODE REVIEW

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (a8ci18n) [inline packages/launch/src/focused-launch/summary/index.tsx:92]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **7** times:
`translate( '2. Confirm your domain' )` *ES Score: 9*

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (a8ci18n) [inline packages/launch/src/focused-launch/summary/index.tsx:145]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **7** times:
`translate( '3. Confirm your plan' )` *ES Score: 9*

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (tjcafferkey) [inline packages/launch/src/focused-launch/summary/index.tsx:103]: Is there a reason we aren't translating this?

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (tjcafferkey) [inline packages/launch/src/focused-launch/summary/index.tsx:156]: Seems like we should be translating this?

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (tjcafferkey) [inline packages/launch/src/focused-launch/summary/index.tsx:164]: Seems like we should be translating this?

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (tjcafferkey) [inline packages/launch/src/focused-launch/summary/index.tsx:174]: Seems like we should be translating this?

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (tjcafferkey) [inline packages/launch/src/focused-launch/summary/index.tsx:124]: Seems like we should be translating this?

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (tjcafferkey) [inline packages/launch/src/focused-launch/summary/index.tsx:134]: Seems like we should be translating this?

**Review by tjcafferkey — CHANGES_REQUESTED** [review]

**Comment by ciampo** [comment]
@tjcafferkey The untranslated text was introduced by https://github.com/Automattic/wp-calypso/pull/46784

I already opened a separate issue https://github.com/Automattic/wp-calypso/issues/47057 to keep track of the untranslated text. We should fix these issues in a separate PR and keep the focus of this PR on being just a refactor

**Comment by alshakero** [comment]
I put these untranslated strings because I was just trying to focus on the layout back then, and instead of lorem ipsum I decided to throw in the original text. I'll work on them later.

**Comment by tjcafferkey** [comment]
@ciampo okay cool that's fine if we're keeping track of those. I've added a checklist to your original issue to include this referenced file. That might also be worth doing to ensure we don't miss any when we come to close the issue as its description is quite ambiguous. 👍🏻 

Feel free to dismiss my requested changes.

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/index.tsx:262]: This is the only difference in behaviour introduced by this PR.

Previously, if the user has a paid domain, the domain step was still rendered together with the other summary steps (but without a "step index" in the title) — see #47110 

With this PR, if the user has a paid domain, the domain step is shown at the top (as a "disabled step"). The remaining "active" steps are shown below.

**Comment by ciampo** [comment]
Incorporated changes from https://github.com/Automattic/wp-calypso/pull/47110 after it got merged.

PR is ready for a proper review.

@alshakero Please have a close look at this PR, and make sure that all of the changes/fixes that you introduced in #47110 have been kept (with the exception of https://github.com/Automattic/wp-calypso/pull/47125#discussion_r518118936)

This PR splits the summary into steps, which should allow for a more modular approach for the next tasks and for an easier way to define the steps logic required in #46858

File: packages/launch/src/focused-launch/summary/index.tsx
Comment (alshakero) [inline packages/launch/src/focused-launch/summary/index.tsx:262]:
```suggestion
	// Steps that are not interactive (e.g. user has already selected domain/plan)
	const disabledSteps = hasPaidDomain ? [ renderDomainStep ] : [];

	// Steps that require the user interaction
	const activeSteps = hasPaidDomain
		? [ renderSiteNameStep, renderPlanStep ]
		: [ renderSiteNameStep, renderDomainStep, renderPlanStep ];
```

**Review by alshakero — APPROVED** [review]
Absolutely love how this is designed. Such a beauty, thank you.

**Pull request merged**
